core/capabilities/remote/trigger_subscriber.go

```diff
@@ -149,7 +158,10 @@ func (s *triggerSubscriber) UnregisterTrigger(ctx context.Context, request commo
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	close(s.registeredWorkflows[request.Metadata.WorkflowID].callback)
+	state := s.registeredWorkflows[request.Metadata.WorkflowID]
+	if state != nil && state.callback != nil {
+		close(state.callback)
+	}
 	delete(s.registeredWorkflows, request.Metadata.WorkflowID)
```

Comment:
should the delete be guarded too?

state, exists := s.registeredWorkflows...
if state != nil ..
if exists {
delete()
}

Reply:
It shouldn't matter for delete() right?

Reply:
no, it's no strictly necessary. you can ignore